sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/BrowserUtils.java

```diff
@@ -0,0 +1,102 @@
+package com.microsoft.azure.mobile.updates;
+
+import android.app.Activity;
+import android.content.ActivityNotFoundException;
+import android.content.Intent;
+import android.content.pm.ActivityInfo;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+
+import com.microsoft.azure.mobile.utils.MobileCenterLog;
+
+import java.util.List;
+
+import static com.microsoft.azure.mobile.updates.UpdateConstants.LOG_TAG;
+
+/**
+ * Browser utils.
+ */
+class BrowserUtils {
+
+    /**
+     * Scheme used to open URLs in Google Chrome instead of any browser.
+     */
+    @VisibleForTesting
+    static final String GOOGLE_CHROME_URL_SCHEME = "googlechrome://navigate?url=";
+
+    @VisibleForTesting
+    BrowserUtils() {
+    }
+
+
+    /**
+     * Open a URL in the best browser available.
+     *
+     * @param url      url to open.
+     * @param activity activity from which to open browser.
+     */
+    static void openBrowser(@NonNull String url, @NonNull Activity activity) {
+
+        /* Try to force using Chrome first, we want fall back url support for intent. */
+        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(GOOGLE_CHROME_URL_SCHEME + url));
+        try {
+            activity.startActivity(intent);
+        } catch (ActivityNotFoundException e) {
+
+            /* Fall back using a browser but we don't want a chooser U.I. to pop. */
+            MobileCenterLog.debug(LOG_TAG, "Google Chrome not found, pick another one.");
+            intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+            List<ResolveInfo> browsers = activity.getPackageManager().queryIntentActivities(intent, 0);
+            if (browsers.isEmpty()) {
+                MobileCenterLog.error(LOG_TAG, "No browser found on device, abort login.");
+            } else {
+
+                /*
+                 * Check the default browser is not the picker,
+                 * last thing we want is app to start and suddenly asks user to pick
+                 * between 2 browsers without explaining why.
+                 */
+                String defaultBrowserPackageName = null;
+                String defaultBrowserClassName = null;
+                ResolveInfo defaultBrowser = activity.getPackageManager().resolveActivity(intent, PackageManager.MATCH_DEFAULT_ONLY);
+                if (defaultBrowser != null) {
+                    ActivityInfo activityInfo = defaultBrowser.activityInfo;
+                    defaultBrowserPackageName = activityInfo.packageName;
+                    defaultBrowserClassName = activityInfo.name;
+                    MobileCenterLog.debug(LOG_TAG, "Default browser seems to be " + defaultBrowserPackageName + "/" + defaultBrowserClassName);
+                }
+                String selectedPackageName = null;
```

Comment:
Since we don't explain why we open a browser before, we don't want user having to make a choice between 2 browsers. So we try to bypass the picker when there is more than 1 browser and no default was explicitly set before.